api/soap/mc_issue_api.php

```diff
@@ -159,6 +159,11 @@ function mci_issue_set_custom_fields( $p_issue_id, array &$p_custom_fields = nul
 	if( isset( $p_custom_fields ) && is_array( $p_custom_fields ) ) {
 		foreach( $p_custom_fields as $t_custom_field ) {
 
+			# convert the rest api multi-dimensional array into a stdClass object
+			if( is_array($t_custom_field) ) {
+				$t_custom_field = json_decode(json_encode($t_custom_field));
```

Comment:
I did not actually test to confirm the issue with HTTP 500 errors when adding custom field, but if all you need is conversion from *array* to *stdClass* then it's much more efficient to use a typecast:
```php
$t_custom_field = (object)$t_custom_field;
```

Reply:
It's a multi-dimensional array. I can not cast all his levels.